### ISSUE TITLE
Error: Error: spawnSync osascript ENOBUFS

### ISSUE THREAD

**Issue by omaerkhan** [motschel123/Mac-Contact-Sync-Obsidian#1]
Hi, I get the following error when running the Sync command:

```
Error: Error: spawnSync osascript ENOBUFS
    at LoadContactsLogic.get_vcards_string_from_applescript (plugin:sync-contacts-macos:364:13)
    at LoadContactsLogic.loadContacts (plugin:sync-contacts-macos:331:27)
    at Object.callback (plugin:sync-contacts-macos:255:49)
```

I'm able to sync successfully when I reduce the number of contacts in my smart list...
There might be a limit to how much data can be sent from the Contacts app?

I have about 1300 contacts so that might explain why it's running out of space.
When I reduce the "Obsidian" Smart List to just 50 or so contacts, it works fine.

Thank you!

**Comment by motschel123** [comment]
Thanks for the feedback, I'll work on it.

**Comment by motschel123** [comment]
@omaerkhan I just finished and published a new version with an improved AppleScript. 
As I don't have 1000's of contacts I can't test for it, so please try it and let me know how it goes. 👍

**Comment by omaerkhan** [comment]
Thank you - brilliant! This particular issue is resolved now...

Interestingly, I get the following message: "Successfully synced 1235 of 1252" so it seems that for some reason, 17 contacts are not being synced... I'll look into it a bit more to see if I can isolate these 17 contacts and figure out why they're being skipped.

Do you think that an error message might be available where the Plugin itself could report why it's skipping some contacts?
I'll open another issue once I am able to identify why these 17 are being skipped.

**Comment by omaerkhan** [comment]
Closing this issue (oops forgot to click that earlier).

**Comment by omaerkhan** [comment]
@motschel123 - I was able to figure out why the rest of the contacts were not syncing... As well as a few other potential issues...

Not syncing:
* Some contacts had a `/` in their name (for instance if a phone number is shared by multiple people)... I was able to resolve these by replacing the `/` with a `,`.  For these contacts, I was receiving an error message in the console.
* Also, I legitimately have multiple contacts with the same first and last names... So I don't think there's any real solution to these - I was forced to artificially change their names in the Contacts list in order to prevent them from failing to sync.

Additional issues I noticed:
* Renaming a contact doesn't rename the corresponding note in Obsidian.
* Deleting a contact doesn't delete the corresponding note in Obsidian.

These issues aren't critical for me, so I won't open an issue for them unless you'd prefer that I do (in case you'd like to track these).

Thank you.
